configuration/src/main/java/io/airlift/configuration/ConfigurationLoader.java

```diff
@@ -57,9 +60,10 @@ public static Map<String, String> loadPropertiesFrom(String path)
             throws IOException
     {
         Properties properties = new Properties();
-        try (InputStream inputStream = new FileInputStream(path)) {
-            properties.load(inputStream);
-        }
+        String escapedProperties = readAllLines(Path.of(path), UTF_8).stream()
+                .map(line -> line.replaceAll("\\\\,", "\\\\\\\\,"))
```

Comment:
Why are we changing the way that properties are loaded? The intent of this PR is to change how `ConfigurationFactory` parses a value into a list for `List<String>` config properties. The above is effectively changing how properties files are parsed -- they no longer adhere to the Java properties specification.